### ISSUE TITLE
First Contentful Paint: NaN

### ISSUE THREAD

**Issue by mathiasbynens** [paulirish/pwmetrics#99]
With pwmetrics@1.2.4:

```
$ pwmetrics https://mathiasbynens.be/
Launching Chrome
  ✘ Error: Sorry, Visually Complete 85% metric is unavailable.
  ✘ Error: Sorry, First Contentful Paint metric is unavailable.

                       |
First Contentful Paint | NaN
                       |
First Meaningful Paint | -1
                       |
Perceptual Speed Index |                                               1,055
                       |
   Time to Interactive | -1
                       -----------------------------------------------
                        0      Time (ms) since navigation start       1055
```

I haven’t looked into what’s causing this, but it seems to be an issue with `pwmetrics` itself, since `lighthouse` handles the same URL just fine.

<details>

```
$ lighthouse https://mathiasbynens.be/
  Lighthouse CLI Launching Chrome... +0ms
  ChromeLauncher Waiting for browser. +386ms
  ChromeLauncher Waiting for browser... +1ms
  ChromeLauncher Waiting for browser..... +504ms
  ChromeLauncher Waiting for browser....... +505ms
  ChromeLauncher Waiting for browser.......✓ +5ms
  status Initializing… +2s
  status Loading page & waiting for onload +382ms URL, HTTPS, Viewport, ThemeColor, Manifest, Accessibility, ImageUsage, ContentWidth
  statusEnd Loading page & waiting for onload +2s
  status Retrieving trace +2ms
  status Retrieving network records +608ms
  status Retrieving: URL +3ms
  status Retrieving: HTTPS +0ms
  status Retrieving: Viewport +2ms
  status Retrieving: ThemeColor +7ms
  status Retrieving: Manifest +2ms
  status Retrieving: Accessibility +224ms
  status Retrieving: ImageUsage +127ms
  status Retrieving: ContentWidth +3ms
  status Loading page & waiting for onload +347ms ServiceWorker, Offline
  statusEnd Loading page & waiting for onload +655ms
  status Retrieving network records +0ms
  status Retrieving: ServiceWorker +2ms
  status Retrieving: Offline +0ms
  status Loading page & waiting for onload +334ms HTTPRedirect, HTMLWithoutJavaScript
  statusEnd Loading page & waiting for onload +925ms
  status Retrieving network records +0ms
  status Retrieving: HTTPRedirect +2ms
  status Retrieving: HTMLWithoutJavaScript +2ms
  status Loading page & waiting for onload +356ms ChromeConsoleMessages, Styles, CSSUsage, EventListeners, AnchorsWithNoRelNoopener, AppCacheManifest, ConsoleTimeUsage, DateNowUse, DocWriteUse, GeolocationOnStart, NotificationOnStart, DOMStats, OptimizedImages, TagsBlockingFirstPaint, WebSQL
  statusEnd Loading page & waiting for onload +716ms
  status Retrieving network records +1ms
  status Retrieving: ChromeConsoleMessages +1ms
  status Retrieving: Styles +4ms
  status Retrieving: CSSUsage +156ms
  status Retrieving: EventListeners +6ms
  status Retrieving: AnchorsWithNoRelNoopener +90ms
  status Retrieving: AppCacheManifest +3ms
  status Retrieving: ConsoleTimeUsage +3ms
  status Retrieving: DateNowUse +2ms
  status Retrieving: DocWriteUse +3ms
  status Retrieving: GeolocationOnStart +2ms
  status Retrieving: NotificationOnStart +16ms
  status Retrieving: DOMStats +11ms
  status Retrieving: OptimizedImages +4ms
  status Retrieving: TagsBlockingFirstPaint +7ms
  status Retrieving: WebSQL +5ms
  status Disconnecting from browser... +505ms
  status Evaluating: Uses HTTPS +31ms
  status Evaluating: Redirects HTTP traffic to HTTPS +0ms
  status Evaluating: Registers a Service Worker +1ms
  status Evaluating: Responds with a 200 when offline +0ms
  status Evaluating: Has a `<meta name="viewport">` tag with `width` or `initial-scale` +2ms
  status Evaluating: Manifest's `display` property is set +1ms
  status Evaluating: Contains some content when JavaScript is not available +0ms
  status Evaluating: First meaningful paint +0ms
  status Evaluating: Perceptual Speed Index +23ms
  status Evaluating: Estimated Input Latency +377ms
  status Evaluating: Time To Interactive (alpha) +245ms
  status Evaluating: User Timing marks and measures +1ms
  status Evaluating: Screenshots of all captured frames +1ms
  status Evaluating: Critical Request Chains +97ms
  status Evaluating: Manifest exists +1ms
  status Evaluating: Manifest contains `background_color` +1ms
  status Evaluating: Manifest contains `theme_color` +0ms
  status Evaluating: Manifest contains icons at least 192px +0ms
  status Evaluating: Manifest contains icons at least 144px +1ms
  status Evaluating: Manifest contains `name` +1ms
  status Evaluating: Manifest contains `short_name` +0ms
  status Evaluating: Manifest's `short_name` won't be truncated when displayed on homescreen +0ms
  status Evaluating: Manifest contains `start_url` +0ms
  status Evaluating: Has a `<meta name="theme-color">` tag +0ms
  status Evaluating: Content is sized correctly for the viewport +1ms
  status Evaluating: Avoids deprecated APIs +0ms
  status Evaluating: Element aria-* attributes are allowed for this role +0ms
  status Evaluating: Elements with ARIA roles have the required aria-* attributes +0ms
  status Evaluating: Element aria-* attributes have valid values +1ms
  status Evaluating: Element aria-* attributes are valid and not misspelled or non-existent. +0ms
  status Evaluating: Background and foreground colors have a sufficient contrast ratio +0ms
  status Evaluating: Every image element has an alt attribute +0ms
  status Evaluating: Every form element has a label +9ms
  status Evaluating: No element has a `tabindex` attribute greater than 0 +0ms
  status Evaluating: Avoids enormous network payloads +0ms
  status Evaluating: Unused CSS rules +5ms
  status Evaluating: Unoptimized images +2ms
  status Evaluating: Oversized Images +2ms
  status Evaluating: Avoids Application Cache +6ms
  status Evaluating: Avoids an excessive DOM size +0ms
  status Evaluating: Opens external anchors using rel="noopener" +1ms
  status Evaluating: Avoids requesting the geolocation permission on page load +0ms
  status Evaluating: Render-blocking Stylesheets +1ms
  status Evaluating: Avoids `console.time()` in its own scripts +0ms
  status Evaluating: Avoids `Date.now()` in its own scripts +1ms
  status Evaluating: Avoids `document.write()` +3ms
  status Evaluating: Avoids Mutation Events in its own scripts +0ms
  status Evaluating: Avoids old CSS flexbox +3ms
  status Evaluating: Avoids WebSQL DB +2ms
  status Evaluating: Avoids requesting the notification permission on page load +0ms
  status Evaluating: Render-blocking scripts +0ms
  status Evaluating: Uses HTTP/2 for its own resources +0ms
  status Evaluating: Uses passive listeners to improve scrolling performance +3ms
  Printer html output written to /private/tmp/mathiasbynens.be_2017-04-19_17-22-39.report.html +322ms
  CLI:warn Report output no longer defaults to stdout. Use `--output=pretty` to re-enable. +1ms
  CLI Protip: Run lighthouse with `--view` to immediately open the HTML report in your browser +1ms

  ChromeLauncher Killing all Chrome Instances +57ms
```

</details>

**Comment by denar90** [comment]
@mathiasbynens hi. Thanks for pointing this out. Can I ask you try the latest master (`npm i -g paulirish/pwmetrics`). I'm not sure about `1.2.4` because we went far far from it in master...

**Comment by mathiasbynens** [comment]
$ npm i -g paulirish/pwmetrics

$ pwmetrics https://mathiasbynens.be/
module.js:471
    throw err;
    ^

Error: Cannot find module '../lib/index'
    at Function.Module._resolveFilename (module.js:469:15)
    at Function.Module._load (module.js:417:25)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (~/.nvm/versions/node/v6.9.1/lib/node_modules/pwmetrics/bin/cli.js:10:19)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)

**Comment by denar90** [comment]
Looks like `ts` files were not built during install from master. We decided not to store built files in repo, just on npm. So yeah, `npm i -g paulirish/pwmetrics` will not help.
@mathiasbynens if you go into installed repo and run `npm run build`, everything should work as expected. Sorry for this kind of pain :(
We are ready for publishing v2, so I ping @paulirish for publishing it.

**Comment by mathiasbynens** [comment]
I actually tried a fresh clone, and then `npm i && npm run build`, but that failed too:

```
lib/index.ts(217,20): error TS2352: Type 'Socket' cannot be converted to type 'TermWritableStream'.
  Property 'columns' is missing in type 'Socket'.

npm ERR! Darwin 16.5.0
npm ERR! argv "/Users/mathias/.nvm/versions/node/v6.9.1/bin/node" "/Users/mathias/.nvm/versions/node/v6.9.1/bin/npm" "run" "build"
npm ERR! node v6.9.1
npm ERR! npm  v3.10.8
npm ERR! code ELIFECYCLE
npm ERR! pwmetrics@1.2.4 build: `tsc`
npm ERR! Exit status 2
npm ERR!
npm ERR! Failed at the pwmetrics@1.2.4 build script 'tsc'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the pwmetrics package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     tsc
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs pwmetrics
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls pwmetrics
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /private/tmp/pwmetrics/npm-debug.log

npm ERR! Darwin 16.5.0
npm ERR! argv "/Users/mathias/.nvm/versions/node/v6.9.1/bin/node" "/Users/mathias/.nvm/versions/node/v6.9.1/bin/npm" "i"
npm ERR! node v6.9.1
npm ERR! npm  v3.10.8
npm ERR! code ELIFECYCLE
npm ERR! pwmetrics@1.2.4 prepublish: `npm run build`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the pwmetrics@1.2.4 prepublish script 'npm run build'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the pwmetrics package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     npm run build
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs pwmetrics
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls pwmetrics
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /private/tmp/pwmetrics/npm-debug.log

> pwmetrics@1.2.4 build /private/tmp/pwmetrics
> tsc

lib/index.ts(217,20): error TS2352: Type 'Socket' cannot be converted to type 'TermWritableStream'.
  Property 'columns' is missing in type 'Socket'.

npm ERR! Darwin 16.5.0
npm ERR! argv "/Users/mathias/.nvm/versions/node/v6.9.1/bin/node" "/Users/mathias/.nvm/versions/node/v6.9.1/bin/npm" "run" "build"
npm ERR! node v6.9.1
npm ERR! npm  v3.10.8
npm ERR! code ELIFECYCLE
npm ERR! pwmetrics@1.2.4 build: `tsc`
npm ERR! Exit status 2
npm ERR!
npm ERR! Failed at the pwmetrics@1.2.4 build script 'tsc'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the pwmetrics package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     tsc
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs pwmetrics
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls pwmetrics
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /private/tmp/pwmetrics/npm-debug.log
```

No time to look into it any further right now so I’ll just wait for v2 :)

**Comment by denar90** [comment]
Thx, I'll look at it later today.

**Comment by pedro93** [comment]
@mathiasbynens thank you for posting this problem. Could you provide us with your software's specs? Version of Node & NPM, Chrome and what terminal you're using?

**Comment by mathiasbynens** [comment]
```sh
$ node -v
v6.9.1

$ npm -v
3.10.8

Google Chrome	57.0.2987.133 (Official Build) (64-bit)
Revision	ec33cd0c06881d919ac0de419d829ad914e0be8f-refs/branch-heads/2987@{#887}
OS	Mac OS X 
```

I’m using iTerm2.app but that shouldn’t matter.

**Comment by pedro93** [comment]
Could you update NodeJS to latest and retry? With the update to typescript we started using asyncs and awaits which may be the reason it's a problem.

**Comment by mathiasbynens** [comment]
Done. Same problem with Node.js v7.8.0 and npm v4.2.0.

**Comment by pedro93** [comment]
It is strange that neither I nor @denar90 came across this bug before. I'm currently not in a position to investigate this further until late next week but I think it is an upstream missing feature see https://github.com/DefinitelyTyped/DefinitelyTyped/issues/13508 for more info.

**Comment by elzii** [comment]
I can confirm having the same issue as @pedro93. 

Running node v7.1.0 and npm v4.0.5

However, cloning the repo and installing/building with yarn seems to work. Steps taken after cloning:

* `yarn`
* `yarn prepublish`
* `yarn build`

**Comment by paulirish** [comment]
the initial issue is fixed in pwmetrics 1.3.1, now out. 

i forked the typescript issue into https://github.com/paulirish/pwmetrics/issues/101

yarn ftw, i guess.


cheers guys.
